cdap-data-fabric/src/main/java/co/cask/cdap/data2/datafabric/dataset/service/DatasetInstanceHandler.java

```diff
@@ -303,7 +337,11 @@ private boolean dropDataset(Id.DatasetInstance datasetInstaceId, DatasetSpecific
       return false;
     }
 
-    opExecutorClient.drop(datasetInstaceId, implManager.getTypeInfo(spec.getType()), spec);
+    DatasetTypeMeta typeMeta = getTypeInfo(datasetInstaceId.getNamespace(), spec.getType());
```

Comment:
not part of your PR, but 'datasetInstaceId' is not spelled correctly (should be datasetInstanceId)